web/src/views_util.js

```diff
@@ -29,6 +29,17 @@ export const COMMON_DROPDOWN_WIDGET_PARAMS = {
     unique_id_type: dropdown_widget.DATA_TYPES.STRING,
     hide_search_box: true,
     bold_current_selection: true,
+    on_mount_callback(instance) {
+        setTimeout(() => {
+            // We let the dropdown widget render to fit content but if the content is more
+            // than max-width, we hide overflown text.
+            // TODO: Figure out a pure CSS solution for this.
+            const $list_item_text = $(instance.popper)
+                .find(".dropdown-list-item-common-styles")
+                .children();
+            $list_item_text.addClass("text-overflow-ellipsis-dropdown-widget");
+        }, 0);
+    },
```

Comment:
I am not sure what's going on here, but I observe the text fitting for a fraction of a second and then live-updating to have ellipses; I suspect this code.

Reply:
Yeah, this appraoch didn't work very well, so I just opted to make this a fixed width and add `text-overflow: ellipsis` to anything that overflows.